Gordon360/Controllers/HousingController.cs

```diff
@@ -229,4 +235,463 @@ public ActionResult<ApartmentApplicationViewModel[]> GetAllApartmentApplication(
+    /// <returns>The created RA_Assigned_Ranges object</returns>
+    [HttpPost("roomrange/assign-ra")]
+    [StateYourBusiness(operation = Operation.ADD, resource = Resource.HOUSING_RA_ASSIGNMENT)]
+    public async Task<ActionResult<RA_Assigned_Ranges>> AssignRaToRoomRange([FromBody] RA_AssignmentViewModel model)
+    {
+        try
+        {
+            var assignedRange = await housingService.AssignRaToRoomRangeAsync(model.Range_ID, model.Ra_ID);
+            return Created("RA assigned to room range successfully.", assignedRange);
+        }
+        catch (InvalidOperationException ex)
+        {
+            return BadRequest(ex.Message);
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Retrieves the list of all assignments.
+    /// </summary>
+    /// <returns>Returns a list of all assignments</returns>
+    [HttpGet]
+    [Route("roomrange/assignment/all")]
+    [StateYourBusiness(operation = Operation.READ_ALL, resource = Resource.HOUSING_RA_ASSIGNMENT)]
+    public async Task<IActionResult> GetRangeAssignments()
+    {
+        try
+        {
+            var RangeAssignments = await housingService.GetRangeAssignmentsAsync();
+            if (RangeAssignments == null)
+            {
+                return NotFound("No Assigned Ranges");
+            }
+            return Ok(RangeAssignments);
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Deletes an RA range assignment
+    /// </summary>
+    /// <param name="rangeId">The Room range of the assignment to delete</param>
+    /// <returns> Returns if completed</returns>
+    [HttpDelete("roomrange/assignment/{rangeId}")]
+    [StateYourBusiness(operation = Operation.DELETE, resource = Resource.HOUSING_RA_ASSIGNMENT)]
+    public async Task<IActionResult> DeleteAssignment(int rangeId)
+    {
+        try
+        {
+            var success = await housingService.DeleteAssignmentAsync(rangeId);
+            if (success)
+            {
+                return Ok(new { message = "Assigment deleted successfully." });
+            }
+            else
+            {
+                return BadRequest(new { message = "Failed to delete the Assignment." });
+            }
+        }
+        catch (ResourceNotFoundException ex)
+        {
+            return NotFound(new { message = ex.ExceptionMessage });
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, $"Internal server error: {ex.Message}");
+        }
+    }
+
+    /// <summary>
+    /// Retrieve the RD of the resident's hall based on their hall ID.
+    /// </summary>
+    /// <param name="hallId">The ID of the hall.</param>
+    /// <returns>Returns the RD's details if found, otherwise null.</returns>
+    [HttpGet("rd/{hallId}")]
```

Comment:
The idiomatic route for this would be `GET /housing/RDs?hallId={hallId}`

Reply:
added to new branch